k8s/errors.go

```diff
@@ -115,3 +115,15 @@ type ErrSnapshotDataFailed struct {
 func (e *ErrSnapshotDataFailed) Error() string {
 	return fmt.Sprintf("SnapshotData %v has failed. Cause: %v", e.ID, e.Cause)
 }
+
+// ErrFailedToApplyCustomSpec error type when CRD objects does not applied successfully
+type ErrFailedToApplyCustomSpec struct {
```

Comment:
Should be `ErrFailedToValidateCustomSpec`, since the spec is already applied, but failing to reach Ready Status

Reply:
Also add type to the structure so that we know what object is failing.